### PR TITLE
New GUI DFT search fields

### LINKED ISSUE #1: New GUI DFT search fields
In https://nomad-lab.eu/prod/v1/staging/gui/search/calculations filters/method/DFT  add a new section (visually identical with what we have now for "Relativity Method" or "Core Electron Treatment") for:

- [ ] Van der Waals correction: the corresponding field, https://nomad-lab.eu/prod/v1/staging/gui/analyze/metainfo/nomad.datamodel.datamodel.EntryArchive/results/method/simulation/dft/van_der_Waals_method

- [ ] Number of spin channels
https://nomad-lab.eu/prod/v1/staging/gui/analyze/metainfo/nomad.datamodel.datamodel.EntryArchive/run/method/electronic/n_spin_channels

Minor changes to the elasticsearch might be needed, corresponding code lives in nomad/datamodel/results.py

It probably would be easiest to find the code for "Relativity Method" or "Core Electron Treatment" searches, find the corresponding commits/merge requests with git blame and just mirror that.

For this to be 100% usable it might need a parser changes, for example for the vdW:
- to check status of vdW parsing at least in the most relevant parsers (Wien2k, QE, VASP, OpenMX, exciting) if this is actually parsed now
- to check that parsers that parse vdW settings now also correctly insert empty string so that we can distinguish between no vdW correction and "unavailable", meaning no parser support.
- But this is not directly related and concerns the parsers instead.

### PR DESCRIPTION
**Pull request by jfrodlova**
* spin polarization
* number of spin channels
* van der Waals method

Fixes ondracka/nomad#1

### CODE REVIEW

File: .gitmodules
Comment (ondracka) [inline .gitmodules:1]: Please put this and the electronic parsers update in a separate commit.

File: nomad/datamodel/results.py [hunk 1]
```diff
@@ -1726,13 +1726,22 @@ class DFT(MSection):
             Elasticsearch(suggestion='default'),
         ],
     )
+
```
Comment (ondracka) [inline nomad/datamodel/results.py]: Unrelated whitespace change.

File: gui/src/components/search/FilterRegistry.js [hunk 2]
```diff
@@ -292,6 +292,9 @@ registerFilter('results.method.simulation.precision.basis_set', idPrecision, {..
 registerFilter('results.method.simulation.precision.planewave_cutoff', idPrecision, {...numberHistogramQuantity, label: 'Plane-wave Cutoff', scale: '1/2'})
 registerFilter('results.method.simulation.precision.apw_cutoff', idPrecision, {...numberHistogramQuantity, label: 'APW Cutoff', scale: '1/2'})
 registerFilter('results.method.simulation.dft.core_electron_treatment', idDFT, termQuantity)
+registerFilter('results.method.simulation.dft.van_der_Waals_method', idDFT, termQuantity)
+registerFilter('results.method.simulation.dft.spin_polarized', idDFT, {...termQuantity, label: 'Spin polarization'})
```
Comment (ondracka) [inline gui/src/components/search/FilterRegistry.js]: Please split into three commit per the specific added search field.

**Review by ondracka — COMMENTED** [review]
This look nice, few comments inlined, but I would still like some polishing.

One issue is that I only see false in the Spin polarization search field, could you please check where the True got lost?
Screenshot from 2024-02-05 10-11-23

Also for the van_der_Waals_method the default value of empty string indicating that there is no correction is kinda stupid for the search field, so we should check for it and replace with "none" and also we could save "unavailable" if the field was not parsed at all (like it works for example for the Core Electron Treatment field a bit above).

**Comment by ondracka** [comment]
Hi Jitka, this is looking good, thanks.

**Comment by jfrodlova** [comment]
Thanks. Anyway, that vDW field is not finished yet. I'll work on it now.

**Comment by ondracka** [comment]
Ah, ****, right I have missed that its still showing the empty value. I'll rebase my develop branch to latest upstream-devel and reapply everything except for the unfinished vdW commit, please open a new MR for that one. (I also managed to merge the OpenMX vdW changes upstream in the meantime, so we can go back to vanilla electronic parsers module).

**Comment by jfrodlova** [comment]
About that vDW search field:

- filter should display "none" in case there is no value in metadata
- uploading new data - should I modify the parser so that it generates "unavailable", when there is no value in the file?

Do I get it right?

**Comment by ondracka** [comment]
> About that vDW search field:
> 
>     * filter should display "none" in case there is no value in metadata

We should display none if there is no correction applied, which currently equals to empty string  from parser. I discussed with developers that having empty string as none is stupid and we should change the metainfo definition to have the parsers output "none" or "no" if there is no vdW correction, but right now we just have to workaround this to not have an empty string in the GUI.

>     * uploading new data - should I modify the parser so that it generates "unavailable", when there is no value in the file?

I think it would be better to do this in the datamodel:
nomad/datamodel/metainfo/simulation/method.py
we probably should just put default value `default=unavailable` to the `van_der_waals_method`

EDIT: pushed enter too early...

**Comment by ondracka** [comment]
Just to make it clear, there are three scenarios:
1) no vdW correction is applied, this now equals to empty string in the metainfo -> should show none in the gui search field
2) some correction was applied, should show the correct method key, i.e., G06 or G10
3) parser does not support parsing of this info (or its not printed in the output), the metainfo field is `None` (this is the python https://www.w3schools.com/python/ref_keyword_none.asp meaning it doesn't exist), here we should show this as unavailable in the GUI, so that we know we should improve the parsers later.
